File: 卡码/代码随想录/Python/图论/dfs/LeetCode_0797.py
#797. 所有可能的路径
from typing import *

class Solution:
	def __init__(self):
		self.res:List[List[int]]=[]
		self.path:List[int]=[]
	def dfs(self,graph:List[List[int]],index:int):
		if index==len(graph)-1:
			self.res.append(self.path[:])
			return
		for i in range(len(graph[index])):
			self.path.append(graph[index][i])
			self.dfs(graph,graph[index][i])
			self.path.pop()
	def allPathsSourceTarget(self, graph: List[List[int]]) -> List[List[int]]:
		self.path.append(0)
		self.dfs(graph,0)
		return self.res

File: 卡码/代码随想录/JS/图论/dfs/LeetCode_0797.js
//797. 所有可能的路径
const allPathsSourceTarget = function(graph) {
	const res=new Array();
	const path=new Array();
	const dfs=(index)=>{
		if(index==graph.length-1){
			res.push(Array.from(path));
			return;
		}
		for(let i=0;i<graph[index].length;i++){
			path.push(graph[index][i]);
			dfs(graph[index][i]);
			path.pop();
		}
	}
	path.push(0);
	dfs(0);
	return res;
};
